wire-ios-sync-engine/Source/SessionManager/SessionManager.swift

```diff
@@ -811,9 +812,20 @@ public final class SessionManager: NSObject, SessionManagerType {
                     if error != nil {
                         self.delegate?.sessionManagerDidFailToLoadDatabase()
                     } else {
-                        let userSession = self.startBackgroundSession(for: account,
-                                                                         with: coreDataStack)
-                        completion(userSession)
+                        let userSession = self.startBackgroundSession(for: account, with: coreDataStack)
+
+                        if self.cryptoboxMigrationManager.isNeeded(in: coreDataStack.accountContainer) {
+                            /// We need to migrate the existing proteus sessions, prekeys, and identity key to CoreCrypto keystore.
+                            self.delegate?.sessionManagerWillMigrateAccount {
+                                userSession.syncContext.performAndWait {
+                                    try? self.cryptoboxMigrationManager.perform(in: coreDataStack.accountContainer,
```

Comment:
I suggest as a first implementation to fail hard in case of an error, making sure that we can easily see why it failed. I suggest to do this with a `do catch` and an assertion, possibly some logs too.

Reply:
Done